bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerService.kt

```diff
@@ -79,7 +84,16 @@ class BusinessPartnerService(
         saveChangelog(resolutionResults)
 
         val partners = resolutionResults.map { it.businessPartner }
-        partners.forEach { entity -> initSharingState(entity) }
+        val orchestratorBusinessPartnersDto = resolutionResults.map { orchestratorMappings.toBusinessPartnerGenericDto(it.businessPartner) }
+        partners.forEach { entity ->
+            initSharingState(entity)
+        }
+
+        val taskCreateResponse = requestNewCleaning(orchestratorBusinessPartnersDto)
+
+        partners.forEach { entity ->
+            updateSharingState(entity, taskCreateResponse.createdTasks[partners.indexOf(entity)])
```

Comment:
That's alright logic here but a bit more optimized you could have an indexed for each, so that the partners list does not need to be searched in every loop iteration.
```suggestion
     partners.withIndex().forEach { (index, entity) ->
            updateSharingState(entity, taskCreateResponse.createdTasks[index])
        }
```